src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1591 @@
+						//Calculate the ventilation mass flow rate
+						Mvent = Msa * OSAF;
+
+						if (StdRhoAir>1) 	pCandidateSetting->Supply_Air_Ventilation_Volume = Mvent / StdRhoAir;
+						else pCandidateSetting->Supply_Air_Ventilation_Volume = Mvent / 1.225;  //Why store it as a volume
+
+						
+						if (Mvent > MinOA_Msa) MinVRMet = MinVRMetOnce = true;
+						else MinVRMet = false;
+
+						if (MinVRMet)
+						{
+							//all these points meet the minimum VR requirement
+							solutionspace->PointMeta[point_number] = 1;
+							// Calculate prospective supply air temperature
+							Tsa = pMode->CalculateCurveVal(1, StepIns.Tosa, Wosa, StepIns.Tra, Wra, Msa, OSAF, modenumber, TEMP_CURVE); //TEMP_CURVE W_CURVE POWER_CURVE
+																														// Check it meets constraints
+							if (MeetsSupplyAirTOC(Tsa)) SAT_OC_Met = SAT_OC_MetOnce = SAT_OC_MetinMode = true;
+							else SAT_OC_Met = false;
+
+							// Calculate prospective supply air Humidity Ratio
+							Wsa = pMode->CalculateCurveVal(1, StepIns.Tosa, Wosa, StepIns.Tra, Wra, Msa, OSAF, modenumber, W_CURVE);
+							//Return Air Relative Humidity(0 - 100 % ) //Return Air Humidity Ratio(g / g)
+							if (MeetsSupplyAirRHOC(Wsa)) SARH_OC_Met = SAHR_OC_MetOnce = SAHR_OC_MetinMode = true;
+							else SARH_OC_Met = false;
+
+
+							if (SARH_OC_Met == true && SAT_OC_Met == true)
+							{
+
+								pCandidateSetting->Mode = pMode->ModeID;
+								pCandidateSetting->Outside_Air_Fraction = OSAF;
+								pCandidateSetting->Supply_Air_Mass_Flow_Rate_Ratio = MsaRatio;
+								pCandidateSetting->Supply_Air_Mass_Flow_Rate = Msa;
+								pCandidateSetting->pMode = pMode;
+								pCandidateSetting->SupplyAirTemperature = Tsa;
+								pCandidateSetting->SupplyAirW = Wsa;
+								pCandidateSetting->Mode = pMode->ModeID;
+								Settings.push_back(pCandidateSetting);
+
+							}
+
+						}
+					}
+				}
+				if (!WarmupFlag)
+				{
+
+					if (SAT_OC_MetinMode == false)
+					{
+						SAT_OC_MetinMode_v[pMode->ModeID] = SAT_OC_MetinMode_v[pMode->ModeID] + 1;	
+					}
+					if (SAHR_OC_MetinMode == false)
+					{
+						SAHR_OC_MetinMode_v[pMode->ModeID] = SAHR_OC_MetinMode_v[pMode->ModeID] + 1;
+					}
+				}
+				modenumber++;
+			}
+
+			std::list<CSetting*>::const_iterator iteratorSetting;
+			for (iteratorSetting = Settings.begin(); iteratorSetting != Settings.end(); ++iteratorSetting) // iterate though the modes.
+			{
+				CSetting * pSetting = *iteratorSetting;
+				//Calculate the delta H 
+				OSAF = pSetting->Outside_Air_Fraction;
+				Msa = pSetting->Supply_Air_Mass_Flow_Rate;
+				Tsa = pSetting->SupplyAirTemperature;
+				modenumber = pSetting->Mode;
+				Wsa = pSetting->SupplyAirW;
+				Tma = StepIns.Tra + OSAF * (StepIns.Tosa - StepIns.Tra);
+				Wma = Wra + OSAF * (Wosa - Wra);
+				pSetting->Mixed_Air_Temperature = Tma;
+				pSetting->Mixed_Air_W = Wma;
+				Hma = 1.006 * Tma * (2501 + 1.86 * Tma); //why was it like this?????
```

Comment:
I don't know why it was like this, but these lines need to be validated by you before we accept them, if you are so uncertain about them.